src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/domain/goal/service/GoalService.kt

```diff
@@ -26,4 +28,26 @@ class GoalService(
     log.info { "Saving Goal [${goal.reference}] for prisoner [$prisonNumber]" }
     return persistenceAdapter.createGoal(goal, prisonNumber)
   }
+
+  /**
+   * Returns a [Goal] identified by its `prisonNumber` and `goalReference`.
+   * Throws [GoalNotFoundException] if the [Goal] cannot be found.
+   */
+  fun getGoal(prisonNumber: String, goalReference: UUID): Goal {
+    log.info { "Retrieving Goal with reference [$goalReference] for prisoner [$prisonNumber]" }
+    return persistenceAdapter.getGoal(prisonNumber, goalReference)
+      ?: throw GoalNotFoundException(prisonNumber, goalReference).also {
+        log.info { "Goal with reference [$goalReference] for prisoner [$prisonNumber] not found" }
+      }
+  }
+
+  /**
+   * Updates a [Goal] identified by its `prisonNumber` and `goalReference`.
+   * Throws [GoalNotFoundException] if the [Goal] cannot be found.
+   */
+  fun updateGoal(prisonNumber: String, goalReference: UUID): Goal {
+    log.info { "Updating Goal with reference [$goalReference] for prisoner [$prisonNumber]" }
+    return getGoal(prisonNumber, goalReference)
```

Comment:
At the moment the "update goal" method only retrieves the Goal from persistence so we can implement the "goal not found scenarios"
Next PRs will flesh this out to actually perform the update